network/p2p/node_sampler.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	safemath "github.com/ava-labs/avalanchego/utils/math"
+	"github.com/ava-labs/avalanchego/utils/sampler"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/version"
+)
+
+const (
+	maxValidatorSetStaleness = time.Minute
+)
+
+var (
+	_ NodeSampler = (*Validators)(nil)
+
+	_ validators.Connector = (*Peers)(nil)
+	_ NodeSampler          = (*Peers)(nil)
+)
+
+type NodeSampler interface {
```

Comment:
Is this `NodeSampler` or `PeerSampler`

Reply:
Seems more like a node sampler to me. It samples nodes (which may or may not be connected)

Reply:
Yeah my thinking was that the terminology we use are:
- `node`: a member in the network
- `peer`: a node we are connected to
- `validator`: a staking node